packages/js/src/executors/node/node.impl.ts

```diff
@@ -73,21 +86,13 @@ export async function* nodeExecutor(
 
   // Re-map buildable workspace projects to their output directory.
   const mappings = calculateResolveMappings(context, options);
-  let outputFileName =
-    buildOptions.outputFileName || `${path.parse(buildOptions.main).name}.js`;
-
-  if (!buildOptions.outputFileName) {
-    const matches = buildOptions.main.match(/^(?!.*src)(.*)\/([^/]*)$/); //ignore strings that contain src and split paths into [folders before main, main.ts]
-    if (matches) {
-      const [mainFolder, mainFileName] = matches.slice(1);
-      outputFileName = path.join(
-        mainFolder,
-        `${path.parse(mainFileName).name}.js`
-      );
-    }
-  }
+  const fileToRun = getFileToRun(
+    context,
+    project,
+    buildOptions,
+    buildTargetOptions
```

Comment:
Why both, here, and not only `buildOptions`, since `buildOptions` contains `buildTargetOptions` too?